SignalServiceKit/src/Messages/Interactions/TSInteraction.m

```diff
@@ -98,7 +98,7 @@ - (instancetype)initInteractionWithTimestamp:(uint64_t)timestamp inThread:(TSThr
 {
     OWSAssertDebug(timestamp > 0);
 
-    self = [super initWithUniqueId:nil];
+    self = [super initWithUniqueId:[[NSUUID UUID] UUIDString]];
```

Comment:
So these uniqueIds aren't compatible with the uniqueIds returned by `getAndIncrementMessageIdWithTransaction`?  Isn't the an issue?

Reply:
I'm not sure what you mean by "compatible" they're all strings. 

The issue is that the `incremented id` doesn't get assigned until the item is saved.

But because there is now a circular reference between an interaction and its attachments, we need the id *before* it's been saved.

Reply:
If we're going to _sometimes_ use UUIDs for interaction ids, let's migrate to always using them... and remove `getAndIncrementMessageIdWithTransaction`.

Reply:
> If we're going to sometimes use UUIDs for interaction ids, let's migrate to always using them... and remove getAndIncrementMessageIdWithTransaction.

SGTM, I'll make that change.